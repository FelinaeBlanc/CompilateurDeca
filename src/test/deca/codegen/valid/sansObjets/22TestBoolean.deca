{
    boolean v = true, f = false;
    boolean x = !!!!!!!!!f;
    boolean m = ((8 >= 9) || (8 < 4) || (4+8+9 > 1)) && (9-1+8-8 > 4);

    if (x){
        println("Vrai");
    }else{
        println("Faux");
    }

    x = !!!!!!!!!v;
    if (x){
        println("Vrai");
    }else{
        println("Faux");
    }

    if ( (5 >= 9) || true ){
        println("Vrai");
    }else{
        println("Faux");
    }

    if(  10 >= 10 && 10 >= 1 && 10 <= 10 && 10 <= 11 && 10 == 10 && 10 > 5 && 10 < 15 && -10 <= 0 && !!true && !!!false  ){
        println("Vrai");
    }else{
        println("Faux");
    }

    if (false || true){
        println("Vrai");
    }else {
        println("Faux");
    }

    if ( !(false || true) ){
        println("Vrai");
    }else {
        println("Faux");
    }
    
    if ( !(false || false) ){
        println("Vrai");
    }else {
        println("Faux");
    }

    
    if ( m ){
        println("Vrai");
    }else {
        println("Faux");
    }

}