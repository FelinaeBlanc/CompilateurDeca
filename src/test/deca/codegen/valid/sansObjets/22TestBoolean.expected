Vrai
Faux
Vrai
Vrai
Vrai
Faux
Vrai
Vrai
